ebos/eclcpgridvanguard.hh

```diff
@@ -108,9 +109,73 @@ public:
     EclCpGridVanguard(Simulator& simulator)
         : EclBaseVanguard<TypeTag>(simulator)
     {
+        this->checkConsistency();
         this->callImplementationInit();
     }
 
+    /*!
+     * Checking consistency of simulator
+     */
+    void checkConsistency()
+    {
+        const auto& runspec = this->eclState().runspec();
+        const auto& config = this->eclState().getSimulationConfig();
+        const auto& phases = runspec.phases();
+
+        // check for correct module setup
+        if (config.isThermal()) {
+            if (getPropValue<TypeTag, Properties::EnableEnergy>() == false) {
+                throw std::runtime_error("Input spesify energy while simulator has disabled try xxx_energy");
+            }
+        } else {
+            if (getPropValue<TypeTag, Properties::EnableEnergy>() == true) {
+                throw std::runtime_error("Input spesify no energy while simulator has energy try xxx_energy");
+            }
```

Comment:
I am not sure if these error messages are understandable to the users. 
Note small typo (spesify -> specify).